### PR TITLE
Add an operation description to locks and log it in etcd.

### LINKED ISSUE #455: Add optional operation description to locks and expose in sf-client

### PR DESCRIPTION
**Pull request by mikalstill**
Fixes #455 except for the client side.

### CODE REVIEW

File: shakenfist/db.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def allocate_network(netblock, provide_dhcp=True, provide_nat=False, name=None,
     ipm = ipmanager.NetBlock(netblock)
     etcd.put('ipmanager', None, net_id, ipm.save())
 
-    with etcd.get_lock('vxlan', None, 'all'):
+    with etcd.get_lock('vxlan', None, 'all', opertaion='Allocate network'):
```
Comment (mandoonandy) [inline shakenfist/db.py:106]: lol. I think we need some unit tests.

**Review by mandoonandy — CHANGES_REQUESTED** [review]
Typo is the main problem.

You mentioned variable naming a couple of days ago (one of the "two most difficult coding problems"). Would you consider changing `operation` to `op` to reduce verbosity?

File: shakenfist/db.py (hunk 1 above)
Comment (mikalstill) [reply shakenfist/db.py:106]: Lies! Also yes.

**Review by mandoonandy — APPROVED** [review]
Thanks. LGTM

**Pull request merged**